torch/distributed/distributed_c10d.py

```diff
@@ -1583,7 +1583,7 @@ def _new_process_group_helper(
             break
 
         # Process group wrapper initialization for supported PGs when TORCH_DISTRIBUTED_DEBUG is set
-        if backend_str in [Backend.GLOO, Backend.NCCL, Backend.UCC]:
+        if backend_str in [Backend.GLOO, Backend.NCCL, Backend.UCC] or backend_str.upper() in Backend._plugins:
```

Comment:
how do we logically know which backends can work with this wrapper?  can we safely allow it to work with all backend plugins?  what happens if one plugin does not support the wrapper- will we get a clean error?

Reply:
I read the ProcessGroupWrapper code， find that communication_op in ProcessGroupWrapper is just communication_op in origin_backend + runCollectiveChecks in gloo, like allreduce:
https://github.com/pytorch/pytorch/blob/82e0153487c2cd1abc92598963be5b57ab1948d4/torch/csrc/distributed/c10d/ProcessGroupWrapper.cpp#L406-L411

`runCollectiveChecks` is used to `collective finger print` for tensors and run gloo's `monitoredBarrier`. 
https://github.com/pytorch/pytorch/blob/82e0153487c2cd1abc92598963be5b57ab1948d4/torch/csrc/distributed/c10d/ProcessGroupWrapper.cpp#L586-L590
So, I think if torch compile with gloo, custom backend can work for ProcessGroupWrapper.